### PR TITLE
Add repository_row_connection controller and actions [SCI-9672]

### PR DESCRIPTION
**Pull request by wandji20**
Jira ticket: [SCI-9672](https://scinote.atlassian.net/browse/SCI-9672)

### What was done
_Add repository_row_connection controller and actions_

#### ToDo:
_N/A_

### Note:
_N/A_

[SCI-9672]: https://scinote.atlassian.net/browse/SCI-9672?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: app/controllers/repository_row_connections_controller.rb [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+  def create
+    connection_params[:relation_ids]
+    RepositoryRowConnection.transaction do
+      @repository.repository_rows.where(id: connection_params[:relation_ids]).find_each do |row|
+        attributes = {
+          created_by: current_user,
+          last_modified_by: current_user,
+          "#{connection_params[:relation]}": row
+        }
+        @repository_row.public_send("#{connection_params[:relation]}_connections").build attributes
```
Comment (okriuchykhin) [inline app/controllers/repository_row_connections_controller.rb:49]: don't use send with user provided params

File: app/controllers/repository_row_connections_controller.rb [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+  before_action :load_repository, except: %i(repositories create)
+  before_action :load_create_vars, only: :create
+  before_action :check_read_permissions, except: :repositories
+  before_action :load_repository_row, except: %i(repositories repository_rows)
+  before_action :check_manage_permissions, except: %i(repositories repository_rows index)
+  skip_before_action :verify_authenticity_token
+
+  def index
+    parents = @repository_row.parent_connections
+                             .joins("INNER JOIN repository_rows ON repository_rows.id = \n
```
[the remark this answers is not in the document]
Comment (wandji20) [reply app/controllers/repository_row_connections_controller.rb:14]: @okriuchykhin I tried using parent_repository_rows and child_repository_rows but would not get the connection id which may be used for unlinking the relation.

File: app/controllers/repository_row_connections_controller.rb [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+                                    repository_row_connections.parent_id")
+                             .select(:id, 'repository_rows.id AS repository_row_id',
+                                     'repository_rows.name AS repository_row_name')
+                             .map do |row|
+                               {
+                                 id: row.id,
+                                 name: row.repository_row_name,
+                                 code: "#{Repository::ID_PREFIX}#{row.repository_row_id}"
```
Comment (okriuchykhin) [inline app/controllers/repository_row_connections_controller.rb:22]: Why are you attaching inventory prefix to the row?

File: app/controllers/repository_row_connections_controller.rb [hunk 4]
```diff
@@ -0,0 +1,125 @@
...
+    children = @repository_row.child_connections
+                              .joins("INNER JOIN repository_rows ON repository_rows.id = \n
+                                     repository_row_connections.child_id")
+                              .select(:id, 'repository_rows.id AS repository_row_id',
+                                      'repository_rows.name AS repository_row_name')
+                              .map do |row|
+                                {
+                                  id: row.id,
+                                  name: row.repository_row_name,
+                                  code: "#{Repository::ID_PREFIX}#{row.repository_row_id}"
```
Comment (okriuchykhin) [inline app/controllers/repository_row_connections_controller.rb:34]: Why are you attaching inventory prefix to the row?

File: app/controllers/repository_row_connections_controller.rb [hunk 5]
```diff
@@ -0,0 +1,125 @@
...
+  end
+
+  def create
+    connection_params[:relation_ids]
+    RepositoryRowConnection.transaction do
+      @repository.repository_rows.where(id: connection_params[:relation_ids]).find_each do |row|
+        attributes = {
+          created_by: current_user,
+          last_modified_by: current_user,
+          "#{connection_params[:relation]}": row
```
Comment (okriuchykhin) [inline app/controllers/repository_row_connections_controller.rb:47]: also don't use params for constructing attribute name

File: app/controllers/repository_row_connections_controller.rb (hunk 2 above)
[the remark this answers is not in the document]
Comment (okriuchykhin) [reply app/controllers/repository_row_connections_controller.rb:14]: Ok, but remove the newline symbol, we don't use them

File: app/controllers/repository_row_connections_controller.rb [hunk 6]
```diff
@@ -0,0 +1,125 @@
...
+  end
+
+  def destroy
+    connection = @repository_row.parent_connections.or(@repository_row.child_connections).find(params[:id])
+    connection.destroy
+    head :no_content
+  end
+
+  def repositories
+    repositories = Repository.active
```
Comment (okriuchykhin) [inline app/controllers/repository_row_connections_controller.rb:74]: You can use here Repository.accessible_by_teams(current_team)

File: app/controllers/repository_row_connections_controller.rb [hunk 7]
```diff
@@ -0,0 +1,125 @@
...
+  def check_read_permissions
+    render_403 unless can_read_repository?(@repository)
+  end
+
+  def check_manage_permissions
+    render_403 unless can_manage_repository_rows?(@repository)
+  end
+
+  def check_view_all_permissions
+    render_403 unless can_read_team?(current_team)
```
Comment (okriuchykhin) [inline app/controllers/repository_row_connections_controller.rb:119]: Not needed

**Review by okriuchykhin — CHANGES_REQUESTED** [review]
Check comments

File: app/controllers/repository_row_connections_controller.rb [hunk 8]
```diff
@@ -0,0 +1,122 @@
+# frozen_string_literal: true
+
+class RepositoryRowConnectionsController < ApplicationController
+  before_action :load_repository, except: %i(repositories create)
+  before_action :load_create_vars, only: :create
+  before_action :check_read_permissions, except: :repositories
+  before_action :load_repository_row, except: %i(repositories repository_rows)
+  before_action :check_manage_permissions, except: %i(repositories repository_rows index)
+  skip_before_action :verify_authenticity_token
```
Comment (okriuchykhin) [inline app/controllers/repository_row_connections_controller.rb:9]: why it is here?

**Review by okriuchykhin — APPROVED** [review]

**Pull request merged**